featuretools/entityset/relationship.py

```diff
@@ -108,7 +108,8 @@ def _is_unique(self):
         """Is there any other relationship with same parent and child entities?"""
         es = self.child_entity.entityset
         relationships = es.get_forward_relationships(self._child_entity_id)
-        n = len([r for r in relationships if r.parent_entity == self.parent_entity])
+        n = len([r for r in relationships
+                 if r._parent_entity_id == self._parent_entity_id])
```

Comment:
This case wasn't causing the observed performance drop, but I also don't see any reason to compare the actual entities here.